### PR TITLE
DEV-57 initial jenkins pipeline

### PR DESCRIPTION
**Pull request by minid33**
Using Jenkins pipelines allows us to have a built-in method of versioning our jenkins jobs with the branch. 

Wherever the Jenkinsfile exists in the root of the branch its contents will be executed on every change (we can obviously write code based on branches /whatever to augment or limit this).

It will allow us to remove duplication inside the jenkins-job-dsl repo. 

This commit is running unit tests and checkstyle. More analyses and better reporting will come in later changes.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 test:
-	mvn -T16 clean install cobertura:cobertura
+	docker run --rm -vi ${CURDIR}:/repo qstack-cloudstack-build /bin/bash -c "cd /repo && MAVEN_OPTS='-Dmaven.repo.local=/repo/.m2' mvn -T16 install cobertura:cobertura"
```
Comment (ProjectMoon) [inline Makefile]: Is the point of the makefile only going to be for running tests and validation?

Comment (minid33) [inline Makefile]: yes at this point, the goal is to test every commit

Comment (ProjectMoon) [inline Makefile]: Looks fine to me, although I haven't downloaded and run this myself.

Comment (minid33) [inline Makefile]: This is where it ran

https://jenkins.qstack.com/job/test-cloudstack-validation/branch/DEV-57-InitialJenkinsPipeline-gq/

Comment (ProjectMoon) [inline Makefile]: 404 not found.

**Comment by ProjectMoon** [comment]
Looks fine now, though you probably need to rebase it before merging.

**Pull request merged**